bug: success rate bellow 100% in nightly test

Nightly tests are failing to start successfully:

https://github.com/freenet/freenet-core/actions/runs/21054646236/job/60548117152

```
Run echo "Running medium scale simulation (50 nodes, 2000 events)"
Running medium scale simulation (50 nodes, 2000 events)
   Compiling freenet v0.1.97 (/home/runner/actions-runner/_work/freenet-core/freenet-core/crates/core)
   Compiling fdev v0.3.71 (/home/runner/actions-runner/_work/freenet-core/freenet-core/crates/fdev)
    Finished `release` profile [optimized] target(s) in 1m 07s
     Running `target/release/fdev test --name nightly-medium-50 --seed 0xDEADBEEF --gateways 4 --nodes 46 --events 2000 --ring-max-htl 12 --max-connections 20 --min-connections 6 --min-success-rate 1.0 --print-summary single-process`

thread 'tokio-runtime-worker' (1115408) panicked at /home/runner/actions-runner/_work/freenet-core/freenet-core/crates/core/src/transport/connection_handler.rs:1076:31:
task shouldn't panic: JoinError::Cancelled(Id(1964))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: found disconnected nodes
Error: Process completed with exit code 1.
```

--- a/crates/core/src/node/testing_impl.rs
+++ b/crates/core/src/node/testing_impl.rs
@@ -3377,6 +3377,7 @@ impl SimNetwork {
         max_contract_num: usize,
         iterations: usize,
         simulation_duration: Duration,
+        event_wait: Duration,
     ) -> anyhow::Result<()>
     where
         R: RandomEventGenerator + Send + 'static,
@@ -3573,7 +3574,7 @@ impl SimNetwork {
                 }
 
                 // Wait between events (Turmoil handles this deterministically)
-                tokio::time::sleep(Duration::from_millis(200)).await;
+                tokio::time::sleep(event_wait).await;
             }
 
             // Wait for events to fully propagate through the network

--- a/crates/fdev/src/testing/single_process.rs
+++ b/crates/fdev/src/testing/single_process.rs
@@ -20,13 +20,21 @@ pub(super) fn run(config: &super::TestConfig) -> anyhow::Result<(), super::Error
     let seed = config.seed();
     let max_contract_num = config.max_contract_number.unwrap_or(config.nodes * 10);
     let iterations = config.events as usize;
+    let event_wait = Duration::from_millis(config.event_wait_ms.unwrap_or(200));
+
+    // Calculate simulation duration from event timing:
+    // total virtual time = startup(2s) + events * wait + propagation(2s) + convergence(10s) + 20% buffer
+    let event_time_secs = (iterations as u64 * event_wait.as_millis() as u64) / 1000;
+    let base_duration_secs = 2 + event_time_secs + 2 + 10;
+    let simulation_duration = Duration::from_secs(base_duration_secs + base_duration_secs / 5);
 
     std::thread::spawn(move || {
         simulated_network.run_fdev_test::<rand::rngs::SmallRng>(
             seed,
             max_contract_num,
             iterations,
-            Duration::from_secs(300), // 5 minute simulation timeout
+            simulation_duration,
+            event_wait,
         )
     })
     .join()

